### PR TITLE
Adds new maintenance spawn

### PR DESCRIPTION
**Pull request by cacogen**
## About The Pull Request

https://user-images.githubusercontent.com/25089914/155257039-d27eda80-d1b0-459c-a29d-67cba9c92f79.mp4

https://user-images.githubusercontent.com/25089914/155257195-2a22abda-0a9c-4f39-be49-54251530d504.mp4

Adds a system similar to smuggler's satchels that spawns five pairs of quantum pads hidden underneath the floor in both maintenance and secluded public areas. 

The pads try to be at least 50 tiles apart from each other to maximise their usefulness.

Unlike smuggler's satchels, they do not show up to observers or on t-ray scanners. Instead, they have a couple of quirks in the way they spawn that make them fairly easy to find. You can always brute force it, too.

## Why It's Good For The Game

It's fun. It makes maintenance less linear. It makes it easier to lose pursuers in maintenance.

## Changelog

:cl:
add: Adds new maintenance spawn
/:cl:

### CODE REVIEW

**Comment by Time-Green** [comment]
I love this so much, but isn’t it a bit common? I think it’d be more fitting as a roundstart trait so it doesn’t get old

**Comment by MrDoomBringer** [comment]
If it makes maintenance objectively more interesting and needs to be actively searched for, I don't see any need to make it any less common than ~3 pairs per round.

Also @cacogen I suggest updating your changelog, as I don't think players will understand what you mean by 'maintenance structure spawner', even if you're intentionally being vague.

File: code/game/objects/structures/ladders.dm
Comment (tralezab) [inline code/game/objects/structures/ladders.dm:209]: this won't be called unless Initialize returns INITIALIZE_HINT_LATELOAD

File: code/game/objects/structures/ladders.dm
Comment (tralezab) [inline code/game/objects/structures/ladders.dm:243]: call spawn_tile() instead and set floor_tile as what it returns

File: code/game/objects/structures/ladders.dm
Comment (tralezab) [inline code/game/objects/structures/ladders.dm:229]: switch this around and early return after ghosts attempt this, you have a small else case at the end of this if proc that makes you have to indent everything

File: code/game/objects/structures/ladders.dm
Comment (tralezab) [inline code/game/objects/structures/ladders.dm:241]: this is a pretty scary assumption that the destination's turf will be a floor. if it's not, this code will break when it tries spawning the floortile that nonfloors don't have.

**Review by tralezab — CHANGES_REQUESTED** [review]
these are strictly the code-side stuff, i have some lame gamey things to comment on as well

**Comment by tralezab** [comment]
ladders teleporting people across the map feels too gamey to me. any chance these could be rethemed as old abandoned warp pads, or whatever you can think up that makes sense?

**Comment by D4C-420** [comment]
I also second the idea of not using ladders because as is currently ladders are used strictly for moving between Z-levels. Not across which might confuse players upon first using them. Best suggestion I have would actually be having them be similar to jaunter portals where they teleport you but also make you throw up and stun you for a bit as a sort of cost of using them (also so you can't just rapidly jump in and out of them like ladders, combined with the fact that those portals from the mining tool already move you randomly  across the station Z level like these do working even on station.)

File: code/game/objects/structures/ladders.dm
Comment (cacogen) [reply code/game/objects/structures/ladders.dm:209]: The parent ladder does that (Line 25)

Comment (tralezab) [reply code/game/objects/structures/ladders.dm:209]: 🙏

**Comment by cacogen** [comment]
>I love this so much, but isn’t it a bit common? I think it’d be more fitting as a roundstart trait so it doesn’t get old

It needs to be secretive but practical. Like it needs to give you a good teleport distance (which is not something I've factored in yet) and they need to be hard to find without never being seen. The right amount is based on how difficult they are to find and how valuable they actually are.

The problems I have at the moment is that they always pick the same handful of spots which makes them very easy to metagame and they can teleport you a remarkably shit distance to the point where you feel ripped off and disappointed.

**Comment by MrDoomBringer** [comment]
@cacogen I'm confused. Why not just generate the location of both pairs programmatically/randomly?

**Comment by cacogen** [comment]
That was definitely the first thing I tried, but the results weren't very good. Iended up with ladders beside each other, or in the middle of hallways or inside of windows. 

I could fix the ladders spawning too close to each other or on turfs with dense contents but you'd still end up with ladders in the middle of maintenance corridors. By using spawners you can make sure they only appear in more out of the way places.

File: code/game/machinery/quantum_pad.dm
Comment (cacogen) [inline code/game/machinery/quantum_pad.dm:205]: This stopped working properly. The pads start above the floor and are only hidden when you remove and replace the tile. Don't know how to fix it.

**Comment by cacogen** [comment]
@MrDoomBringer

It's now random again to prevent metagaming the spawners. But I've incorporated the code I made when I changed it to use spawners, meaning the pads will always be a good distance away from each other. 

I've also made sure they won't end up underneath things like windows. They'll still end up in places like the middle of a maintenance corridor but given the scarcity and the distance between pads it shouldn't be too much of an eyesore.

**Comment by cacogen** [comment]
I've been waiting for a very long time for a review of this.

**Comment by MrDoomBringer** [comment]
I've been waiting for a very long time for a review of this.

**Comment by optimumtact** [comment]
probably because you went out of your way to make every maintainer actively dislike you and now you realise you aren't owed a review.

**Comment by cacogen** [comment]
I’ve never thought I was owed a review. And I don’t think the lack of reviews has anything to do with how the maintainers may or may not feel about me.

**Comment by YakumoChen** [comment]
I don't understand the reason T Ray doesn't detect this. Isn't that the whole point of them, to see stuff under floor tiles? It's inconsistent.

**Comment by cacogen** [comment]
I tested this extensively and found them too easy to find. The easier they are to find the less of them you can have because they're too easy to metagame. This way you have to use a bit of deductive reasoning and trial and error to find them.

Ice Box wasn't a good example of a station to show this on though as the maintenance tunnels don't have a lot of floorspace and most of that isn't covered by tiles, so they're very easy to find. They're harder to find on DeltaStation.

There's a lore paper that spawns with them on a random empty tile in maintenance that explains they've been specifically designed to be undetectable by t-ray scanners.

**Comment by cacogen** [comment]
the end of an era

**Pull request closed without merging**
